app/controllers/api/v2/search_controller.rb

```diff
@@ -18,12 +19,22 @@ def index
 
   private
 
+  def validate_search_params!
+    params.require(:q)
+
+    return if user_signed_in?
+
+    render json: { error: 'Search queries for fewer than 5 characters are not supported without authentication' }, status: 401 if params[:q].size < 5
+    render json: { error: 'Search queries pagination is not supported without authentication' }, status: 401 if params[:offset].present?
+    render json: { error: 'Search queries that resolve remote resources are not supported without authentication' }, status: 401 if truthy_param?(:resolve)
```

Comment:
I think you need to explicitly return, as I don't think `render` breaks the control flow.